### DISCUSSION TITLE
Kubo Client - return response?

### DISCUSSION THREAD

**Discussion by tennox** [subconsciousnetwork/noosphere#768] [Design]
I was delighted to find that there is a (seemingly) ready-to-use kubo RPC client, but discovered that the responses are not returned. :disappointed: 
- e.g. [pin_blocks result](https://github.com/subconsciousnetwork/noosphere/blob/main/rust/noosphere-ipfs/src/client/kubo.rs#L84) is just discarded

It would be a lot more useful to me if that weren't the case - is there a reason for that? Is it worth a PR to change it?

**Comment by jsantell** [comment]
Our [IpfsClient](https://github.com/subconsciousnetwork/noosphere/blob/main/rust/noosphere-ipfs/src/client/mod.rs) trait capabilities (and the [KuboClient](https://github.com/subconsciousnetwork/noosphere/blob/main/rust/noosphere-ipfs/src/client/kubo.rs) implementation) were designed for Noosphere's interaction with IPFS, namely syndicating and pinning blocks on IPFS after storing blocks in Noosphere storage.

If the existing functionality works for you, excellent! Though with caveats: as Noosphere is still in early development stages, we may make aggressive breaking changes, or even drop Kubo support altogether opting for another syndication solution. 

I'm hesitant to expand functionality outside of our use case, but could consider something that doesn't add extra overhead, though here specifically, we'd need to collect all the [PinAddResponse](https://docs.rs/ipfs-api-prelude/latest/ipfs_api_prelude/response/struct.PinAddResponse.html) from the multiple requests per `pin_blocks`. As we only pin orphaned blocks, we don't make use of the returned pinned values.
